### PR TITLE
AWS resources management

### PR DESCRIPTION
**Pull request by rpadovani**
Add methods to manage resources on AWS

### CODE REVIEW

File: AwsResources.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        except Exception as e:
+            logging.error('Impossible logging in:')
+            logging.error(e)
+            logging.error(
+                'Please read https://boto3.amazonaws.com/v1/documentation/api/latest/guide/quickstart.html#configuration '
+                'for a proper configuration of the credentials.')  # TODO: write a guide on our wiki
+
+            raise Exception('Critical error, aborting.')
+
+    def create_ebs(self):
```
Comment (alessandrobessi) [inline AwsResources.py]: Missing return type (`response` type)

File: AwsResources.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+                    # TODO: find a more appropriate AMI
+                }
+            ],
+            Owners=['099720109477']  # Canonical
+        )
+
+        sorted_list = sorted(images, reverse=True,
+                             key=lambda i: strptime(i.creation_date, '%Y-%m-%dT%H:%M:%S.%fZ'))
+
+        return [image for image in sorted_list][0].id
```
Comment (alessandrobessi) [inline AwsResources.py:108]: Isn't `[image for image in sorted_list]` the same as `sorted_list`?

File: AwsResources.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+    def create_key_pair(self):
+        response = self.ec2_client.create_key_pair(KeyName=self.key_name)
+        key_file = open(f'/tmp/{self.key_name}.pem', 'w')
+        key_file.write(response['KeyMaterial'])
```
Comment (alessandrobessi) [inline AwsResources.py]: `key_file.close()` is missing. Alternatively, you can use the `with open(...) as f:` syntax.

File: AwsResources.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+            return self.volume.attach_to_instance(
+                Device='/dev/sdk',
+                InstanceId=instance_id
+            )
+        except Exception as e:
+            logging.error('Impossible attaching the EBS volume to the instance:')
+            logging.error(e)
+            return -1
+
+    def detach_ebs_from_instance(self, volume_id):
```
Comment (alessandrobessi) [inline AwsResources.py]: Missing type hints.

File: AwsResources.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+        self.ec2_client.detach_volume(
+            VolumeId=volume_id
+        )
+
+    def delete_instance(self, instance_id):
```
Comment (alessandrobessi) [inline AwsResources.py]: Missing type hints.

**Review by alessandrobessi — CHANGES_REQUESTED** [review]
Please address the proposed changes.

File: AwsResources.py (hunk 2 above)
Comment (rpadovani) [reply AwsResources.py:108]: Nope, I am casting to list: `sorted_list` is an iterator, and it is not subscriptable

File: AwsResources.py (hunk 3 above)
Comment (rpadovani) [reply AwsResources.py]: Good catch, thanks

File: AwsResources.py (hunk 1 above)
Comment (rpadovani) [reply AwsResources.py]: Rewritten a bit and done

File: AwsResources.py (hunk 4 above)
Comment (rpadovani) [reply AwsResources.py]: Done, ty

File: AwsResources.py (hunk 5 above)
Comment (rpadovani) [reply AwsResources.py]: Done, ty

**Review by alessandrobessi — COMMENTED** [review]
OK!

**Review by alessandrobessi — APPROVED** [review]

**Review by alessandrobessi — APPROVED** [review]

**Pull request merged**